internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -315,6 +324,16 @@ public VeniceWriter(
     }
     this.producerGUID = GuidUtils.getGUID(props);
     this.logger = LogManager.getLogger("VeniceWriter [" + GuidUtils.getHexFromGuid(producerGUID) + "]");
+    // Create a thread pool which can have max 2 threads.
+    // Except during VW start and close we expect it to have zero threads to avoid unnecessary resource usage.
+    this.threadPoolExecutor = new ThreadPoolExecutor(
+        2,
+        2,
+        30,
+        TimeUnit.SECONDS,
+        new LinkedBlockingQueue<>(),
+        new DaemonThreadFactory("VW-" + topicName));
+    this.threadPoolExecutor.allowCoreThreadTimeOut(true); // allow core threads to timeout
```

Comment:
Do we need to shutdown the thread pool in all flavors of `close` even for daemon threads?

Reply:
Good point. 

Done